lib/vnet/daemon/client_darwin.m

```diff
@@ -0,0 +1,98 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#include "client_darwin.h"
+
+#import <Foundation/Foundation.h>
+#import <ServiceManagement/ServiceManagement.h>
+
+// VNECopyNSString duplicates an NSString into an UTF-8 encoded C string.
+// The caller is expected to free the returned pointer.
+char *VNECopyNSString(NSString *val) {
+  if (val) {
+    return strdup([val UTF8String]);
+  }
+  return strdup("");
+}
+
+// Returns the label for the daemon by getting the identifier of the bundle
+// this executable is shipped in and appending ".vnetd" to it.
+//
+// The returned string might be empty if the executable is not in a bundle.
+//
+// The filename and the value of the Label key in the plist file and the Mach
+// service of of the daemon must match the string returned from this function.
+NSString *DaemonLabel(void) {
+  NSBundle *main = [NSBundle mainBundle];
```

Comment:
I moved the decision behind whether to use a launch daemon or osascript behind a build tag. This removed the need to perform those weird shenanigans with `codesign --verify` in Go.

However, that's not the end of the issues. It turns out that `[NSBundle mainBundle]` doesn't handle symlinks well either. When executing `/usr/local/bin/tsh vnet` with the tsh symlinked to `/Applications/tsh.app/Contents/MacOS/tsh`, it thinks that the bundle is at `/usr/local/bin`, which is not the case.

I'm going to add one more change. In Go, I'm going to call `os.Executable`, follow any symlinks, and then check if the dir path ends with `Contents/MacOS`. If so, I'm going to pass that as `bundle_path` to `C.RegisterDaemon`. Then I'm going to call [`[NSBundle bundleWithPath]`](https://developer.apple.com/documentation/foundation/nsbundle/1495012-bundlewithpath).